game-frontend/src/components/user-script/monaco-helper.ts

```diff
@@ -18,6 +24,28 @@ export const useMonacoInstance = (): MonacoInstance | undefined => {
   const [monacoInstance, setMonacoInstance] = useState<MonacoInstance>();
   useEffect(() => {
     monaco.init().then(m => {
+      m.languages.typescript.javascriptDefaults.setCompilerOptions({
+        noLib: true,
+        allowNonTsExtensions: true,
+      });
+      m.languages.typescript.javascriptDefaults.addExtraLib(`
+      type Direction = 'UP' | 'DOWN' | 'LEFT' | 'RIGHT';
+
+      declare function move(direction: Direction): void;
+      declare function consume(): void;
+
+      declare const store: any;
```

Comment:
Could you add the correct typings so the intellisense is useful to the user?

Reply:
could you give me an example?

Reply:
Like foods isn't an array of any. It's an array of Food.

Reply:
he wanted an actual example, but anyway..

Reply:
```.d.ts
type Unit = { /* our unit typings */ };

declare const userUnit: Unit;
```